baybe/recommenders/pure/bayesian/base.py

```diff
@@ -43,16 +46,38 @@ def _validate_deprecated_argument(self, _, value) -> None:
                 "The parameter has been renamed to 'acquisition_function'."
             )
 
+    @property
+    def surrogate_model(self) -> SurrogateProtocol:
+        """Deprecated!"""
+        warnings.warn(
+            f"Accessing the surrogate model via 'surrogate_model' has been "
+            f"deprecated. Use '{self.get_surrogate.__name__}' instead to get the "
+            f"trained model instance (or "
+            f"'{fields(type(self))._surrogate_model.name}' to access the raw object).",
+            DeprecationWarning,
+        )
+        return self._surrogate_model
+
+    def get_surrogate(
+        self,
+        searchspace: SearchSpace,
+        objective: Objective,
+        measurements: pd.DataFrame,
+    ) -> SurrogateProtocol:
+        """Get the trained surrogate model."""
+        self._surrogate_model.fit(searchspace, objective, measurements)
```

Comment:
a comment here that the fit is "cached" and not necessarily triggered would be helpful

Reply:
could you take care of it for me, please? thx 👏🏼

Reply:
added